### PR TITLE
feat(di): change the params of Provider and provide to start with "use"

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

**Comment by vicb** [comment]
I'm -1 on `useExisting`, too vague.

A new comer can easily think that he could use an existing class, value or factory.

```typescript
class Ford extends Car {

}

provide(Car, {useExisting: Ford}); // easy error
```

I would use `useToken`.

**Comment by PascalPrecht** [comment]
:+1: for using `useToken`

File: modules/angular2/test/public_api_spec.ts [hunk 1]
```diff
@@ -109,18 +109,22 @@ var NG_API = [
   'Binding',
   'Binding.dependencies',
   'Binding.multi',
+  'Binding.useExisting',
```
Comment (IgorMinar) [inline modules/angular2/test/public_api_spec.ts]: we are we adding this to binding?

File: modules/angular2/test/public_api_spec.ts [hunk 2]
```diff
@@ -109,18 +109,22 @@ var NG_API = [
...
+  'Binding.useClass',
+  'Binding.useFactory',
+  'Binding.useValue',
+  'Binding.token',
   'Binding.toAlias',
   'Binding.toClass',
   'Binding.toFactory',
   'Binding.toValue',
-  'Binding.token',
```
Comment (IgorMinar) [inline modules/angular2/test/public_api_spec.ts]: is this legit?

Comment (vsavkin) [inline modules/angular2/test/public_api_spec.ts]: This is an issue with git. Binding.token is still here.

**Comment by IgorMinar** [comment]
just to follow up on `useExisting` vs `useToken`, we feel that both are a bit ambiguous in different ways. let's stick to `useExisting` for now and if we hear see a lot of confusion about this we can consider renaming. This feature should be rarely used, so as long as the docs clearly explain what it should be used for, I don't think we should waste too much time on the best naming. There are more frequently used apis that could use our attention instead.

**Comment by PascalPrecht** [comment]
SG to me

**Pull request closed without merging**